webapp/channels/src/components/channel_bookmarks/channel_bookmarks_create_modal.tsx

```diff
@@ -576,6 +609,98 @@ const msg = defineMessages({
-    linkInvalid: {id: 'channel_bookmarks.create.error.invalid_url', defaultMessage: 'Please enter a valid link'},
+    linkInvalid: {id: 'channel_bookmarks.create.error.invalid_url', defaultMessage: 'Please enter a valid link. Could not parse: {link}.'},
     saveError: {id: 'channel_bookmarks.create.error.generic_save', defaultMessage: 'There was an error trying to save the bookmark.'},
 });
+
+const TYPING_DELAY_MS = 250;
+const REQUEST_TIMEOUT = 10000;
+export const useBookmarkLinkValidation = (link: string, onValidated: (validatedLink: string, forced?: boolean) => void) => {
+    const {formatMessage} = useIntl();
+
+    const [loading, setLoading] = useState<URL>();
+    const [error, setError] = useState<ReactNode>();
+    const [suppressed, setSuppressed] = useState(false);
+    const abort = useRef<AbortController>();
+
+    const start = useCallback((url: URL) => {
+        setLoading(url);
+        setError(undefined);
+        setSuppressed(false);
+        abort.current = new AbortController();
+        return abort.current.signal;
+    }, []);
+
+    const cancel = useCallback(() => {
+        abort.current?.abort('stale request');
+        abort.current = undefined;
+        setLoading(undefined);
+    }, []);
+
+    useEffect(() => {
+        const handler = setTimeout(async () => {
+            cancel();
+            if (!link) {
+                return;
+            }
+
+            const url = validHttpUrl(link);
+
+            if (!url) {
+                setError(formatMessage(msg.linkInvalid, {link}));
+                setLoading(undefined);
+                return;
+            }
+
+            const signal = start(url);
+
+            try {
+                await fetch(url, {
+                    mode: 'no-cors',
+
+                    // @ts-expect-error AbortSignal.any exists; remove this directive when next TS upgrade
+                    signal: AbortSignal.any([signal, AbortSignal.timeout(REQUEST_TIMEOUT)]),
+                });
+                onValidated(link);
+            } catch (err) {
+                if (signal === abort.current?.signal) {
+                    setError(continuableLinkErr(url, () => {
+                        onValidated(link, true);
+                        setSuppressed(true);
+                        setError(continuableLinkErr(url));
+                    }));
+                }
+            } finally {
+                setLoading((currentUrl) => {
+                    if (currentUrl !== url) {
+                        // trailing effect of cancelled
+                        return currentUrl;
+                    }
+
+                    return undefined;
+                });
+            }
+        }, TYPING_DELAY_MS);
+
+        return () => clearTimeout(handler);
+    }, [link, start, cancel]);
+
+    return [error, {loading: Boolean(loading), suppressed}] as const;
+};
+
+export const validHttpUrl = (input: string) => {
+    const val = parseLink(input);
```

Comment:
ParseLink adds 'http://' if necessary, but not 'https://' resulting in validation failure.